debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/YugabyteDBConnector.java

```diff
@@ -61,10 +62,70 @@ public void start(Map<String, String> props) {
 
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
+        if (props == null) {
+            return Collections.emptyList();
+        }
+
+        if (props.containsKey("snapshot.mode") && props.get("snapshot.mode").equalsIgnoreCase("parallel")) {
+            LOGGER.info("Initialising parallel snapshot consumption");
+
+            final String tableIncludeList = props.get("table.include.list");
+            // Perform basic validations.
+            if (tableIncludeList == null) {
+                throw new DebeziumException("No table provided, provide a table in the table.include.list");
+            } else if (tableIncludeList.contains(",")) {
+                // This might indicate the presence of multiple tables in the include list, we do not want that.
+                throw new DebeziumException("parallel snapshot consumption is only supported with one table at a time");
+            }
+
+            // Publication auto create mode should now be for all tables.
+            if (props.containsKey("publication.autocreate.mode") && props.get("publication.autocreate.mode").equalsIgnoreCase("all_tables")) {
+                throw new DebeziumException("Snapshot mode parallel is not supported with publication.autocreate.mode all_tables, " +
+                                            "use publication.autocreate.mode=filtered");
```

Comment:
Can we move these validations to a separate method `validateSingleTableProvidedForSnapshot()`.